ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatRecord.java

```diff
@@ -171,6 +171,44 @@ static boolean isValidBlobRecordVersion(short blobRecordVersion) {
   }
 
   /**
+   * Deserialize a complete blob record into a {@link BlobAll} object.
+   * @param stream the {@link InputStream} from which to read the blob record.
+   * @param storeKeyFactory the factory for parsing store keys.
+   * @return a {@link BlobAll} object with the {@link BlobInfo} and {@link BlobData} for the blob.
+   * @throws IOException
+   * @throws MessageFormatException
+   */
+  public static BlobAll deserializeBlobAll(InputStream stream, StoreKeyFactory storeKeyFactory)
+      throws IOException, MessageFormatException {
+    skipHeaders(stream);
```

Comment:
Good point. Although what `MessageSievingInputStream` does may be an overkill? That was introduced because data was being written without deserialization previously during replication - so no checks were being done. I think here most of the validity checks happen as the bytes are deserialized. All that seems to be missing is the header validity check - @cgtz should check `MessageFormatSend::calculateOffsets()` to see what is missing here when compared to a `BlobInfo` call, say.